File: Back-end/user-server/src/main/java/com/lht/pojo/BodyLog.java
package com.lht.pojo;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.Generated;
import org.hibernate.annotations.GenerationTime;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "body_log")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BodyLog {

    @Id
    @Generated(GenerationTime.INSERT)
    @Column(columnDefinition = "uuid")
    private UUID uuid;

    @Column(name = "customer_uuid", nullable = false)
    private UUID customerUuid;

    @Column(nullable = false, precision = 5, scale = 2)
    private BigDecimal weight;

    @Column(nullable = false, precision = 4, scale = 1)
    private BigDecimal height;

    @Column(name = "body_fat_percent", precision = 4, scale = 1)
    private BigDecimal bodyFatPercent;

    @Column(name = "muscle_mass", precision = 5, scale = 2)
    private BigDecimal muscleMass;

    @Column(name = "staff_uuid", nullable = false)
    private UUID staffUuid;

    private String note;

    @Column(name = "logged_at")
    private LocalDateTime loggedAt;

    @PrePersist
    public void prePersist() {
        if (loggedAt == null) loggedAt = LocalDateTime.now();
    }
}

File: Front-end/app/lib/models/body_log.dart
class BodyLog {
  final String? uuid;
  final String customerUuid;
  final String? staffUuid;
  final double weight;
  final double height;
  final double? bodyFatPercent;
  final double? muscleMass;
  final String? note;
  final DateTime? loggedAt;

  BodyLog({
    this.uuid,
    required this.customerUuid,
    this.staffUuid,
    required this.weight,
    required this.height,
    this.bodyFatPercent,
    this.muscleMass,
    this.note,
    this.loggedAt,
  });

  factory BodyLog.fromJson(Map<String, dynamic> json) => BodyLog(
    uuid: json['uuid'],
    customerUuid: json['customerUuid'],
    staffUuid: json['staffUuid'],
    weight: (json['weight'] as num).toDouble(),
    height: (json['height'] as num).toDouble(),
    bodyFatPercent: json['bodyFatPercent'] != null
        ? (json['bodyFatPercent'] as num).toDouble()
        : null,
    muscleMass: json['muscleMass'] != null
        ? (json['muscleMass'] as num).toDouble()
        : null,
    note: json['note'],
    loggedAt: json['loggedAt'] != null
        ? DateTime.tryParse(json['loggedAt'])
        : null,
  );

  Map<String, dynamic> toJson() => {
    'customerUuid': customerUuid,
    'staffUuid': staffUuid,
    'weight': weight,
    'height': height,
    'bodyFatPercent': bodyFatPercent,
    'muscleMass': muscleMass,
    'note': note,
  };
}